crates/ruff_linter/src/rules/pylint/rules/too_many_branches.rs

```diff
@@ -107,6 +107,7 @@ fn num_branches(stmts: &[Stmt]) -> usize {
                     .map(|case| num_branches(&case.body))
                     .sum::<usize>()
             }
+            Stmt::With(ast::StmtWith { body, .. }) => num_branches(body),
```

Comment:
I think it should include the `with` statement _and_ the number of branches in its body.

```suggestion
            Stmt::With(ast::StmtWith { body, .. }) => 1 + num_branches(body),
```

Reply:
I think the number of branches is sufficient because the with item itself doesn't introduce any branching.

Reply:
The count would then be inconsistent with other branch constructs which _do_ include the statement itself in the final count.

Reply:
For the python file below, I get the same amount of branches in both cases from pylint. Seems like pylint doesn't include the `with` statement.

```bash
❯ pylint bad.py
************* Module bad
bad.py:6:8: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
bad.py:4:0: R0912: Too many branches (11/10) (too-many-branches)
bad.py:31:4: R1705: Unnecessary "elif" after "return", remove the leading "el" from "elif" (no-else-return)
bad.py:30:0: R0912: Too many branches (11/10) (too-many-branches)
```

```python
from contextlib import suppress


def num_to_word_with(x):  # [too-many-branches]
    with suppress(Exception):
        if x == 0:
            return 'zero'
        elif x == 1:
            return 'one'
        elif x == 2:
            return 'two'
        elif x == 3:
            return 'three'
        elif x == 4:
            return 'four'
        elif x == 5:
            return 'five'
        elif x == 6:
            return 'six'
        elif x == 7:
            return 'seven'
        elif x == 8:
            return 'eight'
        elif x == 9:
            return 'nine'
        else:
            return None


def num_to_word(x):  # [too-many-branches]
    if x == 0:
        return 'zero'
    elif x == 1:
        return 'one'
    elif x == 2:
        return 'two'
    elif x == 3:
        return 'three'
    elif x == 4:
        return 'four'
    elif x == 5:
        return 'five'
    elif x == 6:
        return 'six'
    elif x == 7:
        return 'seven'
    elif x == 8:
        return 'eight'
    elif x == 9:
        return 'nine'
    else:
        return None
```

Reply:
> The count would then be inconsistent with other branch constructs which do include the statement itself in the final count.

I think that inconsistency is desired because with items don't introduce a new branch. They are not branching instructions.

Reply:
Oh yeah, that makes sense. I'll mark this as resolved then. Thanks @blueraft for looking into pylint.